### PR TITLE
fix(api): updated wreck

### PR DESCRIPTION
**Pull request by christophwitzko**
This PR updates [wreck](https://github.com/hapijs/wreck).

### CODE REVIEW

**Comment by Charlotteis** [comment]
LGTM :+1:

**Comment by janl** [comment]
The question I have is, why would `data` which I expect to be the response body, get a `bearerToken` field? Isn’t this always a HTTP header?

Also do we have a test for this failure?

**Comment by christophwitzko** [comment]
@Charlotteis thank you :relaxed: 
@janl If a client sends a request to the server, then the `Authorization` header field is used. (e.g. https://github.com/hoodiehq/hoodie.js/blob/master/src/hoodie/request.js#L58 and on the server https://github.com/hoodiehq/hoodie-server/blob/master/lib/server/plugins/api/index.js#L13)
But how does the client get the bearer token? I think this is the reason why the server sends the token via JSON. The https://github.com/hoodiehq/hoodie.js library inspects the response and saves the `bearerToken` field.

regarding tests: https://github.com/hoodiehq/hoodie-integration-test @boennemann ?

**Comment by janl** [comment]
>  I think this is the reason why the server sends the token via JSON. The https://github.com/hoodiehq/hoodie.js library inspects the response and saves the bearerToken field.

@gr2m is this correct? I was under the impression that the bearerToken is always communicated via HTTP headers, but I’m happy to be wrong on this.

@christophwitzko: Regarding tests, I meant: https://github.com/hoodiehq/hoodie-server/blob/master/test/unit/api-plugin-test.js

**Comment by gr2m** [comment]
> The question I have is, why would data which I expect to be the response body, get a bearerToken field

the bearer token should only be added to the server response of `POST /_session`. Because that's where we read it out. For all over responses, `.bearerToken` property should not be added to the server response.

**Comment by christophwitzko** [comment]
@gr2m that means we have to check additionally the request method and the path?

**Comment by gr2m** [comment]
> @gr2m that means we have to check additionally the request method and the path?

I guess. I'm not a hapi expert. I can only describe the higher level workflow, which is:

1. I sign in with `POST /_api/_session`
2. I expect a response without cookie headers, but with `bearerToken` in the response JSON
3. I don't want `bearerToken` to be added to any other server response

**Comment by janl** [comment]
@christophwitzko can you get this fixed?

**Comment by christophwitzko** [comment]
@janl Everything should work now. :+1:

File: lib/server/hapi_plugins/api/internals.js [hunk 1]
```diff
@@ -11,6 +11,8 @@ module.exports = {
       return;
     }
 
+    var isSess = request.method === 'post' && request.path === '/_api/_session';
```
Comment (christophwitzko) [inline lib/server/hapi_plugins/api/internals.js]: The condition described in this [comment](https://github.com/hoodiehq/hoodie-server/pull/355#issuecomment-119677979).

**Comment by gr2m** [comment]
looks good to me code wise, didn't test it

**Comment by janl** [comment]
style: I moved the `isSess` declaration closer to where it is used.

Otherwise this is good, thanks!

**Comment by christophwitzko** [comment]
Thank you. :+1: :relaxed:

**Pull request merged**